x-pack/filebeat/tests/integration/status_reporter_test.go

```diff
@@ -0,0 +1,217 @@
+		nullDeviceFile = "NUL"
+	}
+	inputStreamIrregular := getInputStream(unitOneID, nullDeviceFile, 1)
+
+	outputExpectedStream := proto.UnitExpected{
+		Id:             unitOutID,
+		Type:           proto.UnitType_OUTPUT,
+		ConfigStateIdx: 1,
+		State:          proto.State_HEALTHY,
+		Config: &proto.UnitExpectedConfig{
+			DataStream: &proto.DataStream{
+				Namespace: "default",
+			},
+			Type:     "file",
+			Revision: 1,
+			Meta: &proto.Meta{
+				Package: &proto.Package{
+					Name:    "system",
+					Version: "1.17.0",
+				},
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
+				"type":            "file",
+				"enabled":         true,
+				"path":            outPath,
+				"filename":        "beat-out",
+				"number_of_files": 7,
+			}),
+		},
+	}
+
+	observedStates := make(chan *proto.CheckinObserved)
+	expectedUnits := make(chan []*proto.UnitExpected)
+	done := make(chan struct{})
+	// V2 mock server
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			select {
+			case observedStates <- observed:
+				return &proto.CheckinExpected{
+					Units: <-expectedUnits,
+				}
+			case <-done:
+				return nil
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+	}
+	require.NoError(t, server.Start())
+	defer server.Stop()
+
+	// start the client
+	client := client.NewV2(fmt.Sprintf(":%d", server.Port), token, client.VersionInfo{
+		Name: "program",
+		Meta: map[string]string{
+			"key": "value",
+		},
+	}, client.WithGRPCDialOptions(grpc.WithTransportCredentials(insecure.NewCredentials())))
+
+	lbmanagement.SetManagerFactory(func(cfg *conf.C, registry *reload.Registry) (lbmanagement.Manager, error) {
+		c := management.DefaultConfig()
+		if err := cfg.Unpack(&c); err != nil {
+			return nil, err
+		}
+		return management.NewV2AgentManagerWithClient(c, registry, client, management.WithStopOnEmptyUnits)
+	})
+
+	go func() {
+		t.Logf("Running beats...")
+		err := cmd.Filebeat().Execute()
+		require.NoError(t, err)
+	}()
+
+	scenarios := []struct {
+		expectedStatus proto.State
+		nextInputunit  *proto.UnitExpected
+	}{
+		{
+			proto.State_HEALTHY,
+			&inputStreamIrregular,
+		},
+		{
+			proto.State_DEGRADED,
+			&inputStream,
+		},
+		{
+			proto.State_HEALTHY,
+			&inputStream,
+		},
+		// wait for one more checkin, just to be sure it's healthy
+		{
+			proto.State_HEALTHY,
+			&inputStream,
+		},
+	}
+
+	timer := time.NewTimer(2 * time.Minute)
+	id := 0
+	for id < len(scenarios) {
+		select {
+		case observed := <-observedStates:
+			state := extractState(observed.GetUnits(), unitOneID)
+			expectedUnits <- []*proto.UnitExpected{
+				scenarios[id].nextInputunit,
+				&outputExpectedStream,
+			}
+			if state != scenarios[id].expectedStatus {
+				continue
+			}
+			// always ensure that output is healthy
+			outputState := extractState(observed.GetUnits(), unitOutID)
+			require.Equal(t, outputState, proto.State_HEALTHY)
```

Comment:
A little detail here, the way the code is written, the output status is only checked if the current state is the desired state, was that your intention? The comment made me understand you wanted to check the output status on every iteration.

Reply:
> the output status is only checked if the current state is the desired state, was that your intention? 

Yes. I wanted to make sure that output is in healthy state.